ecsact/entt/wrapper/dynamic.hh

```diff
@@ -0,0 +1,200 @@
+	using ecsact::entt::detail::pending_add;
+
+	registry.template emplace_or_replace<pending_add<C>>(entity_id);
+
+	if constexpr(!C::transient) {
+		if(registry.template all_of<component_removed<C>>(entity_id)) {
+			registry.template erase<component_removed<C>>(entity_id);
+		} else {
+			registry.template emplace_or_replace<component_added<C>>(entity_id);
+		}
+	}
+}
+
+template<typename C>
+auto context_remove(
+	ecsact_system_execution_context*          context,
+	[[maybe_unused]] ecsact_component_like_id component_id
+) -> void {
+	assert(ecsact_id_cast<ecsact_component_like_id>(C::id) == component_id);
+
+	using ecsact::entt::component_changed;
+	using ecsact::entt::component_removed;
+	using ecsact::entt::detail::beforeremove_storage;
+	using ecsact::entt::detail::pending_remove;
+
+	auto  entity = context->entity;
+	auto& registry = *context->registry;
+
+	registry.template remove<component_added<C>>(entity);
+	registry.template remove<component_changed<C>>(entity);
+	registry.template emplace_or_replace<pending_remove<C>>(entity);
+	registry.template emplace_or_replace<component_removed<C>>(entity);
+
+	// Stop here (tag)
+	if constexpr(!std::is_empty_v<C>) {
+		auto component = registry.template get<C>(entity);
+
+		auto& remove_storage =
+			registry.template emplace_or_replace<beforeremove_storage<C>>(entity);
+
+		remove_storage.value = component;
+	}
+}
+
+template<typename C>
+auto component_remove_trivial(
+	::entt::registry&       registry,
+	ecsact::entt::entity_id entity_id
+) -> void {
+	using ecsact::entt::component_changed;
+	using ecsact::entt::component_removed;
+	using ecsact::entt::detail::beforeremove_storage;
+	using ecsact::entt::detail::pending_remove;
+
+	registry.template remove<component_added<C>>(entity_id);
+	registry.template remove<component_changed<C>>(entity_id);
+	registry.template emplace_or_replace<pending_remove<C>>(entity_id);
+	registry.template emplace_or_replace<component_removed<C>>(entity_id);
+
+	if constexpr(!std::is_empty_v<C>) {
+		auto component = registry.template get<C>(entity_id);
+
+		auto& remove_storage =
+			registry.template emplace_or_replace<beforeremove_storage<C>>(entity_id);
+
+		remove_storage.value = component;
+	}
+}
+
+template<typename C>
+auto context_get(
+	ecsact_system_execution_context*          context,
+	[[maybe_unused]] ecsact_component_like_id component_id,
+	void*                                     out_component_data
+) -> void {
+	auto        entity = context->entity;
+	const auto& registry = *context->registry;
+
+	assert(registry.template any_of<C>(entity));
+
+	*static_cast<C*>(out_component_data) = registry.template get<C>(entity);
+}
+
+template<typename C>
+auto context_update(
+	ecsact_system_execution_context*          context,
+	[[maybe_unused]] ecsact_component_like_id component_id,
+	const void*                               in_component_data
+) -> void {
+	using ecsact::entt::component_changed;
+	// TODO(Kelwan): for remove, beforeremove_storage
```

Comment:
something unresolved or should this comment be removed?

Reply:
Surprised this showed up for you, I got rid of it before requesting review

Reply:
maybe you didn't push everything? It's still there.